### PR TITLE
featurefile: fix LigatureCaret and subtable breaks impl.

### LINKED ISSUE #2024: fea: invalid subtable breaks
~~~
lookup Substitutionsimplelookup7 {
  lookupflag 0;
    sub \a by \ordfeminine ;
    sub \o by \ordmasculine ;
} Substitutionsimplelookup7;

lookup ordnOrdinauxinLatinlookup5 {
  lookupflag 0;
    sub [\zero \one \two \three \four \five \six \seven \eight \nine ] [\a ]'lookup Substitutionsimplelookup7  ;
  subtable;
    sub [\zero \one \two \three \four \five \six \seven \eight \nine ] [\o ]'lookup Substitutionsimplelookup7  ;
  subtable;
    sub [\zero \one \two \three \four \five \six \seven \eight \nine ] [\period ] [\a ]'lookup Substitutionsimplelookup7  ;
  subtable;
    sub [\zero \one \two \three \four \five \six \seven \eight \nine ] [\period ] [\o ]'lookup Substitutionsimplelookup7  ;
} ordnOrdinauxinLatinlookup5;
~~~

~~~
makeotfexe [WARNING] <CharterNova-Regular> subtable break is supported only in c
lass kerning lookups [CharterNova-Regular.ufo\features.fea 263]
makeotfexe [WARNING] <CharterNova-Regular> subtable break is supported only in c
lass kerning lookups [CharterNova-Regular.ufo\features.fea 265]
makeotfexe [WARNING] <CharterNova-Regular> subtable break is supported only in c
lass kerning lookups [CharterNova-Regular.ufo\features.fea 267]
~~~

### PR DESCRIPTION
**Pull request by adrientetar**
- Parse LigatureCaretByIndex per spec
- Stop outputting subtable breaks where unsupported. Closes #2024.

r? @frank-trampe

### CODE REVIEW

**Comment by frank-trampe** [comment]
@adrientetar, can you explain the first problem in more detail?

**Comment by adrientetar** [comment]
That's just parsing of LigatureCaretByIndex so it's ignored instead of failing.

**Comment by frank-trampe** [comment]
Is there a reason for dropping/inlining the function?

File: fontforge/featurefile.c [hunk 1]
```diff
@@ -5667,13 +5648,8 @@ static void fea_ParseGDEFTable(struct parseState *tok) {
...
 	    }
-	} else if ( strcmp(tok->tokbuf,"LigatureCaret")==0 || /* FF backwards compatibility */ \
-		 /* strcmp(tok->tokbuf,"LigatureCaretByIndex")==0  TODO should include this */ \
-		    strcmp(tok->tokbuf,"LigatureCaretByPos")==0 ) {
-	    /* Older versions of FontForge was using LigatureCaret but there is actually */
-	    /* LigatureCaretByPos and LigatureCaretByIndex which we need to watch (2013) */
-	    /* 2013may16 TODO: We need to update all this featurefile stuff according to */
-	    /* http://www.adobe.com/devnet/opentype/afdko/topic_feature_file_syntax.html */
+	} else if ( strcmp(tok->tokbuf,"LigatureCaretByPos")==0 ) {
+	    // Ligature carets by single coordinate (format 1).
```
Comment (frank-trampe) [inline fontforge/featurefile.c]: Why are we dropping the backwards compatibility?

File: fontforge/featurefile.c [hunk 2]
```diff
@@ -5696,11 +5672,7 @@ static void fea_ParseGDEFTable(struct parseState *tok) {
 	    }
 	    for (;;) {
 		fea_ParseTok(tok);
-		if ( tok->type==tk_int )
-		    /* Not strictly cricket, but I'll accept it */;
-		else if ( tok->type==tk_char && tok->tokbuf[0]=='<' )
-		    fea_ParseCaret(tok);
-		else
```
Comment (frank-trampe) [inline fontforge/featurefile.c]: Is there a reason for us to be more picky here than before?

Comment (adrientetar) [inline fontforge/featurefile.c]: Dunno, I felt it would be good to be able to drop the deprecated syntax. Maybe I can bring it back if needed.

Comment (frank-trampe) [inline fontforge/featurefile.c]: Is the second case actually deprecated?

Comment (adrientetar) [inline fontforge/featurefile.c]: Yes, FontForge has been using <LigatureCaret X> as previous syntax.

Comment (frank-trampe) [inline fontforge/featurefile.c]: Let's keep it until it conflicts with something.

**Comment by adrientetar** [comment]
Updated.

**Comment by frank-trampe** [comment]
The code looks good to me. Have you tested?

**Comment by adrientetar** [comment]
Will do tomorrow.

**Comment by adrientetar** [comment]
Imports/exports as expected.

**Pull request merged**